src/Domain.LinnApps/Requisitions/RequisitionManager.cs

```diff
@@ -389,5 +391,43 @@ public async Task<RequisitionHeader> CreateLoanReq(int loanNumber)
 
             return await this.repository.FindByIdAsync(reqNumber);
         }
+
+        public async Task PickStockOnRequisitionLine(RequisitionHeader header, LineCandidate lineWithPicks)
+        {
+            var line = header.Lines.SingleOrDefault(l => l.LineNumber == lineWithPicks.LineNumber);
+
+            if (line == null)
+            {
+                throw new PickStockException("Could not find line");
+            }
+
+            // if no moves before
+            if (!line.Moves.Any())
+            {
+                if (lineWithPicks.StockPicks != null)
+                {
+                    foreach (var pick in lineWithPicks.StockPicks)
+                    {
+                        var fromLocation = string.IsNullOrEmpty(pick.Location)
+                            ? null
+                            : await this.storageLocationRepository.FindByAsync(x => x.LocationCode == pick.Location);
+                        var pickResult = await this.requisitionStoredProcedures.PickStock(
+                            lineWithPicks.PartNumber,
+                            header.ReqNumber,
+                            lineWithPicks.LineNumber,
+                            pick.Qty,
+                            fromLocation?.LocationId, // todo - do we pass a value here if palletNumber?
```

Comment:
i think the answer to this comment/question is no

Reply:
i think we can do some tests with picking stock from pallets